test/missing.jl

```diff
@@ -0,0 +1,225 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+@testset "MissingException" begin
+    @test sprint(showerror, MissingException("test")) == "MissingException: test"
+end
+
+@testset "convert" begin
+    @test convert(Union{Int, Missing}, 1) === 1
+    @test convert(Union{Int, Missing}, 1.0) === 1
+    @test_throws MethodError convert(Missing, 1)
+    @test_throws MethodError convert(Union{Int, Missing}, "a")
+end
+
+@testset "promote rules" begin
+    @test promote_type(Missing, Missing) == Missing
+    @test promote_type(Missing, Int) == Union{Missing, Int}
+    @test promote_type(Int, Missing) == Union{Missing, Int}
+    @test promote_type(Int, Any) == Any
+    @test promote_type(Any, Any) == Any
+    @test promote_type(Missing, Any) == Any
+    @test promote_type(Any, Missing) == Any
+    @test promote_type(Union{Int, Missing}, Missing) == Union{Int, Missing}
+    @test promote_type(Missing, Union{Int, Missing}) == Union{Int, Missing}
+    @test promote_type(Union{Int, Missing}, Int) == Union{Int, Missing}
+    @test promote_type(Int, Union{Int, Missing}) == Union{Int, Missing}
+    @test promote_type(Any, Union{Int, Missing}) == Any
+    @test promote_type(Union{Int, Missing}, Union{Int, Missing}) == Union{Int, Missing}
+    @test promote_type(Union{Float64, Missing}, Union{String, Missing}) == Any
+    @test promote_type(Union{Float64, Missing}, Union{Int, Missing}) == Union{Float64, Missing}
+    @test_broken promote_type(Union{Void, Missing, Int}, Float64) == Any
```

Comment:
I don't really understand whether this bug is in this PR or reflects a deeper Julia bug. See https://github.com/JuliaData/Missings.jl/issues/47.

Reply:
@JeffBezanson , @vtjnash, it'd be great to have you take a look at this